app/controllers/countries_controller.rb

```diff
@@ -64,7 +64,8 @@ def destroy
   private
     # Use callbacks to share common setup or constraints between actions.
     def set_country
-      @country = Country.find(params[:id])
+      #see https://hackhands.com/ruby-rails-performance-tuning/
+      @country = Country.includes( :percentages => :languages).find(params[:id])
```

Comment:
I am specifying in advance all the associations that are going to be loaded.
I actually moved this to the show action (def show), since it is at the show view where I show the country and its percentages and languages.